code/game/machinery/chemHandC.dm

```diff
@@ -0,0 +1,291 @@
+
+//Heater
+
+/obj/machinery/chemheater
+	name = "directed laser heater"
+	desc = "A platform with an integrated laser that uses high-energy photons to heat a subject through atomic vibrations. In a practical sense, it has no upper limit to how much thermal energy can be induced this way, as it is capable of reaching temperatures which could rapidly destroy any laboratory-approved container."
+	icon = 'icons/obj/chemHandC.dmi'
+	icon_state = "heater"
+	icon_state_open = "heater_open"
+	density = 1
+	anchored = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK | EJECTNOTDEL
+	use_power = 1
+	idle_power_usage = 25
+	active_power_usage = 5000
+
+	var/max_temperature = TEMPERATURE_LASER
+	var/thermal_energy_transfer = 3000
+	var/laser_kind = 0
+	var/onstage = null
+
+	var/obj/item/weapon/reagent_containers/held_container
+	var/heating = FALSE
+	var/had_item = FALSE
+
+/obj/machinery/chemheater/New()
+	. = ..()
+
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/chemheater,
+		/obj/item/weapon/stock_parts/micro_laser,
+		/obj/item/weapon/stock_parts/capacitor
+	)
+	RefreshParts()
+
+/obj/machinery/chemheater/RefreshParts()
+	var/T = 0
+	for(var/obj/item/weapon/stock_parts/micro_laser/L in component_parts)
+		T += L.rating
+		laser_kind = L.rating //Sets what tier of laser we have
+	thermal_energy_transfer = initial(thermal_energy_transfer) * T
+
+	T = 0
+	for(var/obj/item/weapon/stock_parts/capacitor/C in component_parts)
+		T += C.rating-1
+	idle_power_usage = initial(idle_power_usage) - (T * 10) //T1: 25w, T2: 15w, T3: 5w
+	active_power_usage = initial(active_power_usage) - (T * 2000) //T1: 5000w, T2: 3000w, T3: 1000w
+
+	overlays = null
+	overlays += image(icon = icon, icon_state = "t[laser_kind]_laser")
+	update_icon()
```

Comment:
```
/obj/update_icon()
	return
```

This is the code you're running here and in the other 7 occurrences of `update_icon()`

Reply:
Instead of adding the overlays and then updating the icon, you should update the icon and have the update icon proc add the overlays.

Reply:
whatever